### PR TITLE
Maps

### PR DESCRIPTION
**Pull request by jackcarlisle**
# ready for review

adds information on maps to the readme
 
#21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -723,6 +723,83 @@ This is an incredibly powerful tool that comes baked-in with elixir. It means th
...
+# %{legs: 4, name: "Max", type: "dog"}
+```
+
+2. Using syntax  
+We can use a special syntax for updating a map in Elixir. It looks like this:
+```elixir
+iex> %{animals | name: "Max"}
+# %{legs: 4, name: "Max", type: "dog"}
+```
+**NOTE: Unlike the function method above, this syntax can only be used to UPDATE
```
Comment (nelsonic) [inline README.md]: @jackcarlisle same comment as in the previous PR. don't think the _entire_ **NOTE** _needs_ to be **bold**, just the **Note**: ... 👍

Comment (jackcarlisle) [inline README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -3,6 +3,19 @@
...
+- [What?](#what)
+- [How?](#how)
+- [Handy Tips](#handy-tips)
+- [Basic Types](#basic-types)
+- [Functions and Modules](#functions-and-modules)
+- [Generate an Elixir Project](#generating-your-first-elixir-project)
+- [Documentation](#documentation)
+- [Testing](#testing)
+- [Data Structures](#data-structures)
+  - [Maps](#maps)
```
Comment (iteles) [inline README.md]: 👍

File: README.md [hunk 3]
```diff
@@ -118,6 +133,9 @@ type `i` followed by the value name:
 
 ![elixir-i](https://cloud.githubusercontent.com/assets/14013616/20860322/3c01d984-b96d-11e6-8cc4-a46c8657f5b4.png)
 
+***
```
Comment (iteles) [inline README.md]: I think these tend to look a bit weird with the double lines caused by the titles, but not a blocker at all. We don't have these anywhere else so I might suggest we remove them for consistency.

screen shot 2017-02-16 at 18 51 46

File: README.md [hunk 4]
```diff
@@ -394,8 +415,10 @@ will be available (provided that `iex` is started
 iex> Math.sum(1, 2)
 3
 ```
+***
+
 
-## Generating your first Elixir project
+## *Generating your first Elixir project*
```
Comment (iteles) [inline README.md]: Is this meant to italicise this title? Or did you mean to make it a 3 hash title?

File: README.md [hunk 5]
```diff
@@ -636,9 +659,9 @@ def load(filename) do
 end
 ```
 
-#### Pipe Operator. What if we wanted to call some of our functions in succession
-to another. Let's create a function that creates a zoo, randomises it and then
-returns a selected number of animals to go and see:
+#### Pipe Operator
+
+What if we wanted to call some of our functions in succession to another. Let's create a function that creates a zoo, randomises it and then returns a selected number of animals to go and see:
```
Comment (iteles) [inline README.md]: Think there should be a question mark at the end of the first sentence? Is it not a question?

**Comment by jackcarlisle** [comment]
@iteles rebased and ready to go

**Pull request merged**